integration-tests/hibernate-orm-panache-rest/src/test/java/io/quarkus/it/panache/rest/AuthorControllerTest.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.it.panache.rest;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import javax.ws.rs.core.Response;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+class AuthorControllerTest {
+
+    private static final String APPLICATION_HAL_JSON = "application/hal+json";
+
+    private static final Author DOSTOEVSKY = Author.create(1L, "Fyodor Dostoevsky", LocalDate.of(1821, 11, 11));
+
+    private static final Author ORWELL = Author.create(2L, "George Orwell", LocalDate.of(1903, 06, 25));
+
+    @Test
+    void shouldGetOne() {
+        when().get("/authors/" + DOSTOEVSKY.id)
+                .then().statusCode(200)
+                .and().body("name", is(equalTo(DOSTOEVSKY.name)))
+                .and().body("dob", is(equalTo(DOSTOEVSKY.dob.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")))));
+    }
+
+    @Test
+    void shouldGetAll() {
+        when().get("/authors")
+                .then().statusCode(200)
+                .and().body("name[0]", is(equalTo(DOSTOEVSKY.name)))
+                .and().body("dob[0]", is(equalTo(DOSTOEVSKY.dob.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")))))
+                .and().body("name[1]", is(equalTo(ORWELL.name)))
+                .and().body("dob[1]", is(equalTo(ORWELL.dob.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")))));
+    }
+
+    @Test
+    void halGetShouldNotBeExposed() {
+        given().accept(APPLICATION_HAL_JSON)
+                .when().get("/authors/" + DOSTOEVSKY.id)
+                .then().statusCode(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+        given().accept(APPLICATION_HAL_JSON)
+                .when().get("/authors")
+                .then().statusCode(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+    }
+
+    @Test
+    void postShouldNotBeExposed() {
+        given().contentType(APPLICATION_JSON)
+                .and().body(new BookDto.AuthorDto(ORWELL))
+                .when().post("/authors")
+                .then().statusCode(Response.Status.METHOD_NOT_ALLOWED.getStatusCode());
```

Comment:
`404` would mean that the resource was not found. `405` is returned if a method is not available (e.g. DELETE).
With `@PanacheRestResource(exposed=false)` you can hide specific methods of a resource. So for example `/books/{id}` resource could be accessible via `GET` but not via `DELETE`. If a DELETE request is attempted, its response would be `405 Method not allowed`

Reply:
Ah, OK, that's a good point. Do you check that there's at least one method for that URL, though? What if I unexpose all methods for a given URL (GET, UPDATE, DELETE I guess)?

Reply:
No it's not checking for that. But because all the methods are exposed by default, user would have to purposefully do this. We could add it later if it's an issue.